### DISCUSSION TITLE
Specify order of output json

### DISCUSSION THREAD

**Discussion by CaptainDriftwood** [wyfo/apischema#182] [Q&A]
For a dataclass, is there a way to specify the order of the attributes when serializing the object?

**Comment by wyfo** [comment]
By default, *apischema* orders the fields according to their position in class declaration. 
Also, [serialized methods/property] are inserted after, in order to potentially overwrite normal fields, whereas aggregate fields like [properties](https://wyfo.github.io/apischema/json_schema/#with-dataclass) or [flattened](https://wyfo.github.io/apischema/data_model/#composition-over-inheritance-composed-dataclasses-flattening) are inserted before, in order to be overwritten.

But there is no API specifically provided to modify order the fields. On the other hand, it can currently be done using [object serialization](https://wyfo.github.io/apischema/conversions/#object-serialization-select-only-a-subset-of-fields).

Could this solution fit your use case?

By the way, when #181 will be released (I know, I'm very late), it will also be possible to reorder fields by using [`apischema.objects.set_object_fields`](https://wyfo.github.io/apischema/data_model/#dataclass-like-types-aka-object-types).

Of course, reordering fields for serialization will also affect ordering of the associated JSON schema (but you will surely agree that it's a good thing).

I know these solutions are more workarounds, as they were not implemented for this purpose, and I'm open to improvements. I'm for example thinking about adding an "order" metadata [like .NET's one](https://www.newtonsoft.com/json/help/html/JsonPropertyOrder.htm), but aggregate fields make things complicated… 
Feel free to add your suggestions.

**Reply by wyfo** [reply]
Field ordering has been addressed in #192
